validation/mock_trace_validation.go

```diff
@@ -57,3 +62,22 @@ func validateTimeStamps(span *cloudtrace.Span) error {
 	}
 	return nil
 }
+
+func AddSpans(uploadedSpans map[string]*cloudtrace.Span, lock *sync.Mutex, spans ...*cloudtrace.Span) error {
```

Comment:
Good work on picking up on the potential concurrency issues. You might want to look into https://blog.golang.org/race-detector as well.

Reply:
Yep, I'll run tests and build my code with the `-race` flag from now on